### PR TITLE
fix: updated azurerm version and fix workflows

### PR DESCRIPTION
**Pull request by vaibhavcloud**
## what

- azurerm *version change from 2.95.0 to 3.0.0
- renamed _example to examples & added workflows

## why
*  because of changes in azurerm 4.1.0

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -148,7 +120,7 @@ variable "vnet_diff_subs_name" {
 }
 variable "alias_subs_id" {
   type        = string
-  default     = ""
+  default     = "068245d4-3c94-42fe-9c4d-9e5e1cabc60c"
```
Comment (13archit) [inline variables.tf:123]: Please replace this value with one I shared in chat.

Comment (d4kverma) [reply variables.tf]: @vaibhavcloud Please resolve this commit also

File: .github/workflows/tf-checks.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+  workflow_dispatch:
+jobs:
+  tf-checks-azure-managed-example:
+    uses: clouddrove/github-shared-workflows/.github/workflows/tf-checks.yml@master
+    with:
+      working_directory: './examples/'
```
Comment (13archit) [inline .github/workflows/tf-checks.yml:11]: We need to add folders path for tfchecks, Please refer vnet module tfcheks and correct it.

File: variables.tf [hunk 3]
```diff
@@ -148,7 +120,7 @@ variable "vnet_diff_subs_name" {
 }
 variable "alias_subs_id" {
   type        = string
-  default     = ""
+  default     = "000000-11111-1223-XXX-XXXXXXXXXXXX"
```
Comment (13archit) [inline variables.tf:123]: Please remove this, keep empty string.

Comment (vaibhavcloud) [reply variables.tf]: updated

File: .github/workflows/tf-checks.yml (hunk 2 above)
Comment (vaibhavcloud) [reply .github/workflows/tf-checks.yml:11]: added

File: variables.tf (hunk 1 above)
Comment (vaibhavcloud) [reply variables.tf]: resolved

**Pull request merged**